### PR TITLE
Dri2: Drop libXext dependency

### PR DESCRIPTION
**Pull request by siro20**
Untested. Please test on DRI2 enabled hardware.

### CODE REVIEW

File: d3d9-nine/dri2.c [hunk 1]
```diff
@@ -274,6 +276,7 @@ static BOOL dri2_create(Display *dpy, int screen, struct dri_backend_priv **priv
     Window root = RootWindow(dpy, screen);
     drm_auth_t auth;
 
+
```
Comment (dhewg) [inline d3d9-nine/dri2.c]: Leftover?

File: d3d9-nine/dri2.c [hunk 2]
```diff
@@ -711,6 +714,37 @@ static BOOL dri2_probe(Display *dpy)
     struct dri_backend_priv *priv;
     BOOL res;
 
+    xcb_connection_t *conn = XGetXCBConnection(dpy);
+    xcb_dri2_query_version_cookie_t dri2_cookie;
+    xcb_dri2_query_version_reply_t *dri2_reply;
+    xcb_generic_error_t *error;
+    const xcb_query_extension_reply_t *extension;
+    const int major = 1;
+    const int minor = 4; //FIXME
```
Comment (dhewg) [inline d3d9-nine/dri2.c]: What's the FIXME about? We're not sure which version we require?

File: d3d9-nine/dri2.c [hunk 3]
```diff
@@ -711,6 +714,37 @@ static BOOL dri2_probe(Display *dpy)
...
+        return FALSE;
+    }
+
+    dri2_cookie = xcb_dri2_query_version(conn, major, minor);
+
+    dri2_reply = xcb_dri2_query_version_reply(conn, dri2_cookie, &error);
+    if (!dri2_reply)
+    {
+        free(error);
+        WINE_ERR("Issue getting requested v%d.%d of DRI3\n", major, minor);
```
Comment (dhewg) [inline d3d9-nine/dri2.c]: DRI2

File: meson.build [hunk 4]
```diff
@@ -81,6 +81,7 @@ dep_x11 = dependency('x11')
 dep_x11_ext = dependency('xext')
 dep_x11_xcb = dependency('x11-xcb')
 dep_xcb = dependency('xcb')
+dep_xcb_dri2 = dependency('xcb-dri2')
```
Comment (dhewg) [inline meson.build]: I think this can be moved down to the dri2 block, so it doesn't get checked if it's disabled?

Comment (dhewg) [reply meson.build]: And please add it to travis in this commit too, so we keep bisectability

File: d3d9-nine/dri2.c (hunk 2 above)
Comment (siro20) [reply d3d9-nine/dri2.c]: correct. I haven't checked which one is the minimum we need.

File: d3d9-nine/dri2.c (hunk 1 above)
Comment (siro20) [reply d3d9-nine/dri2.c]: yes, I'll remove it

File: meson.build (hunk 4 above)
Comment (dhewg) [reply meson.build]: The build dependency `libxcb-dri2-0-dev` needs to be added, no? In contrast, we don't explicitly list `libxext-dev` there, so we don't need to remove that at least.

Comment (siro20) [reply meson.build]: On fedora it's only one libxcb packet for all libraries.
The packages are already installed as dependency, but listing them here seems a good idea.

File: d3d9-nine/dri2.c (hunk 2 above)
Comment (siro20) [reply d3d9-nine/dri2.c]: we only need 1.0, but must use 1.4 to get the version the xserver supports.

**Comment by dhewg** [comment]
Manually merged with a small fixup ("dep_xcb_dri2 = null_dep")

**Pull request closed without merging**